modules/transport-netty4/src/main/java/org/opensearch/transport/Netty4ModulePlugin.java

```diff
@@ -164,21 +169,53 @@ public Map<String, Supplier<HttpServerTransport>> getSecureHttpTransports(
         SecureHttpTransportSettingsProvider secureHttpTransportSettingsProvider,
         Tracer tracer
     ) {
-        return Collections.singletonMap(
-            NETTY_SECURE_HTTP_TRANSPORT_NAME,
-            () -> new SecureNetty4HttpServerTransport(
-                settings,
-                networkService,
-                bigArrays,
-                threadPool,
-                xContentRegistry,
-                dispatcher,
-                clusterSettings,
-                getSharedGroupFactory(settings),
-                secureHttpTransportSettingsProvider,
-                tracer
-            )
-        );
+        if (Http3Utils.isHttp3Available() == true && SETTING_HTTP_HTTP3_ENABLED.get(settings).booleanValue() == true) {
```

Comment:
Why introduce a feature flag for HTTP3? iirc there wasn't a feature flag for http2.